usr.sbin/bhyve/migration.c

```diff
@@ -698,40 +600,30 @@ migrate_kern_data(struct vmctx *ctx, int socket, enum migration_transfer_req req
 	error = 0;
 	snapshot_kern_structs = get_snapshot_kern_structs(&ndevs);
 
-	buffer = malloc(SNAPSHOT_BUFFER_SIZE * sizeof(char));
+	buffer = malloc(SNAPSHOT_BUFFER_SIZE);
 	if (buffer == NULL) {
-		fprintf(stderr,
-			"%s: Could not allocate memory\r\n",
-			__func__);
+		EPRINTF("Could not allocate memory.");
 		return (-1);
 	}
 
 	for (i = 0; i < ndevs; i++) {
 		if (req == MIGRATION_RECV_REQ) {
-			rc = migrate_recv_kern_struct(ctx, socket, buffer);
+			rc = migrate_kern_struct(ctx, socket, buffer, -1,  MIGRATION_RECV_REQ);
 			if (rc < 0) {
-				fprintf(stderr,
-					"%s: Could not restore struct %s\n",
-					__func__,
-					snapshot_kern_structs[i].struct_name);
+				DPRINTF("Could not restore struct %s", snapshot_kern_structs[i].struct_name);
 				error = -1;
 				break;
 			}
 		} else if (req == MIGRATION_SEND_REQ) {
-			rc = migrate_send_kern_struct(ctx, socket, buffer,
-					snapshot_kern_structs[i].req);
+			rc = migrate_kern_struct(ctx, socket, buffer,
+					snapshot_kern_structs[i].req, MIGRATION_SEND_REQ);
```

Comment:
This and the first `if` could probably be merged, considering only the request parameter differs. Considering it's only used on in the `send` part of the `migrate_kern_struct`, it would be safe to also set it proper for `receive`. It would be probably aid debugging too, when printing the error message.